app/src/main/java/org/greenstand/android/TreeTracker/permissions/Permissions.kt

```diff
@@ -0,0 +1,211 @@
+) {
+    val navController = LocalNavHostController.current
+    val state by viewModel.state.observeAsState(PermissionItemsState())
+    val permissionsState = rememberMultiplePermissionsState(
+        permissions = listOf(
+            Manifest.permission.ACCESS_COARSE_LOCATION,
+            Manifest.permission.ACCESS_COARSE_LOCATION,
+            Manifest.permission.CAMERA,
+        )
+    )
+    val lifecycleOwner = LocalLifecycleOwner.current
+    DisposableEffect(
+        key1 = lifecycleOwner,
+        effect = {
+            val observer = LifecycleEventObserver { _, event ->
+                if (event == Lifecycle.Event.ON_START) {
+                    permissionsState.launchMultiplePermissionRequest()
+                }
+            }
+            lifecycleOwner.lifecycle.addObserver(observer)
+            onDispose {
+                lifecycleOwner.lifecycle.removeObserver(observer)
+            }
+        }
+    )
+
+    permissionsState.permissions.forEach { perm ->
+        when (perm.permission) {
+            Manifest.permission.CAMERA -> {
+                when {
+                    perm.hasPermission -> {
+                        return
+                    }
+                    perm.shouldShowRationale -> {
+                        AlertDialog(
+                            onDismissRequest = { navController.popBackStack() },
+                            title = {
+                                Text(text = stringResource(R.string.accept_camera_permission_header))
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_camera_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            }
+                        )
+                    }
+                    else -> {
+                        PermissionDeniedPermanentlyDialog(navController)
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_FINE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        if(state.isLocationEnabled == false){
+                            enableLocation()
+                        }

```

Comment:
This can be
perm.hasPermission && state.isLocationEnabled -> enableLocation()

Reply:
If I refactor the code to this and location is enabled on the phone the else in the when block will run which shouldn't be so if permissions have been granted. So I think it is still in a way better like this.